docs/sequence.puml

```diff
@@ -0,0 +1,58 @@
+@startuml
+database Bitcoin
+actor Bob
+Actor Alice
+database Monero
+
+Bob --> Alice: Establish connection
+
+group Request Response Channel
+    Bob -> Alice: Request Amounts
+    note left: Btc Amount
+
+    Alice --> Bob: Send Amounts
+    note right: Btc & Xmr Amounts
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message0
+    note left: Pubkeys\ndleq proof sb\nxmr viewkey vb\nbtc refund addr
+
+    Alice --> Bob: alice::Message0
+    note right: Pubkeys\ndleq proof sa\nxmr view key va\nbtc redeem addr\nbtc punish addr
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message1
+    note left: btc lock tx
+
+    Alice --> Bob: alice::Message1
+    note right: cancel tx sig\ntx refund enc sig
+end
+
+group Request Response Channel
+    Bob -> Alice: bob::Message2
+    note left: punish tx sig\ncancel tx sig
+
+== Negotiated ==
+
+    Bob ->> Bitcoin: Lock
+
+    Alice ->> Monero: Lock
+
+    Alice --> Bob: alice::Message2
+    note right: xm transfer proof
+end
+
+    Bob -> Alice: bob::Message3
+    note left: Enc sig
```

Comment:
:thinking: If your intention with the diagram is to reflect the state of the codebase I think this should still be part of a `Request Response Channel` group and `Alice` will send back an empty response. But you probably already know this and it's a deliberate omission.

Reply:
I missed that, thanks!

Reply:
Actually, looking at the codebase it does not seem to be the case. Alice does not seem to reply.

Reply:
I was referring to this line, but maybe it's no longer used: https://github.com/comit-network/xmr-btc-swap/blob/07270df12a6805466a5c1cba12bae9e07fdbe1e8/swap/src/protocol/alice/message3.rs#L81

Reply:
I actually look for that initially and did not find it! Thanks.